### PR TITLE
Feature/navigation drawer

### PR DESCRIPTION
**Pull request by Allen-Muhani**

### CODE REVIEW

File: src/screens/Drawer/CustomDrawer.tsx [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    return (
+      <SafeAreaView style={styles.container}>
+        <View>
+            <LinearGradient
+                colors={[
+                    "rgba(255, 140, 161, 0.08)",
+                    "rgba(252, 207, 47, 0.08)",
+                    "rgba(255, 255, 255, 0.08)",
+                    "rgba(248, 48, 180, 0.08)",
+                    "rgba(47, 68, 252, 0.08)",
```
Comment (steffenkrogmann) [inline src/screens/Drawer/CustomDrawer.tsx]: @Allen-Muhani
How I understand it: You specify colors for the gradient here. on the screen.

We don't want to specifiy any colors, fonts, styles etc on any screen. Those should be defined in a centralized file.
I guess it is in styles/theme (?)
gradients are also sort of a color.

please check with @Chegemartinx and @AlphonceMutebi

File: src/screens/Drawer/CustomDrawer.tsx [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+                          style={styles.phoneCountryFlag}
+                        />
+                        
+                        <Text style={styles.phoneNumberTest}>+254 706 427718</Text> 
+                       
+                    </View>
+                </View>
+
+                <TouchableOpacity onPress={() => props.navigation.closeDrawer()} style={{ marginTop: 20 }}>
+                    <AntDesign name="close" size={24} color="black" />
```
Comment (steffenkrogmann) [inline src/screens/Drawer/CustomDrawer.tsx:49]: @Allen-Muhani @Chegemartinx @AlphonceMutebi there is styling here. how can we centralize that?

File: src/screens/Drawer/CustomDrawer.tsx [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        shadowOpacity: 0.25,
+        shadowRadius: 25,
+        elevation: 100
+      },
+    
+      text: {
+        ...FONTS.h5,
+      },
+    
+      ksh: {
```
Comment (steffenkrogmann) [inline src/screens/Drawer/CustomDrawer.tsx:153]: @Chegemartinx @Allen-Muhani @AlphonceMutebi which font style s this?
It should have to be set to "bold" here. There should be one style

File: src/screens/Drawer/CustomDrawer.tsx [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+        ...FONTS.mediumTitle,
+        color: COLORS.textBlack,
+        fontWeight: 'bold',
```
Comment (steffenkrogmann) [inline src/screens/Drawer/CustomDrawer.tsx:156]: @Chegemartinx @Allen-Muhani @AlphonceMutebi which font style s this?
It should have to be set to "bold" here. There should be one style

File: src/screens/Drawer/CustomDrawer.tsx [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+        margin: 2
+      },
+    
+      cUSD: {
+        ...FONTS.body3,
+        color: COLORS.textBlack, 
+        fontWeight: 'bold',
```
Comment (steffenkrogmann) [inline src/screens/Drawer/CustomDrawer.tsx:163]: @Chegemartinx @Allen-Muhani @AlphonceMutebi which font style s this?
It should have to be set to "bold" here. There should be one style

File: src/screens/Drawer/MyDrawer.tsx [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+
+export default function MyDrawer() {
+  return (
+    <Drawer.Navigator
+      screenOptions={{
+        headerShown: false,
+        drawerStyle: {
+          width: SIZES.width * 0.8,
+          borderRadius: 15,
+          backgroundColor: '#F5F5F5',
```
Comment (steffenkrogmann) [inline src/screens/Drawer/MyDrawer.tsx]: @Chegemartinx @Allen-Muhani there is a background color her. That one should be in a central place how do we do that?

File: src/screens/Drawer/MyDrawer.tsx [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+  // width: SIZES.width * 0.9,
+  marginTop: -5,
+  marginLeft: -10,
+  paddingBottom: 5
+}
+
+const drawerLabelStyle: TextStyle = {
+      fontSize: 18,
+      color: COLORS.primary,
+      fontFamily: "Rubik_400Regular",
```
Comment (steffenkrogmann) [inline src/screens/Drawer/MyDrawer.tsx]: @Chegemartinx @Allen-Muhani there is a fontsize and font family in here. 
Those should be in a central place how do we do that?

File: src/screens/Onboarding/OnboardingItem.tsx [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+        <Text style={styles.description}>{description}</Text>
+      </View>
+    </View>
+  );
+};
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    backgroundColor: "#fff",
```
Comment (steffenkrogmann) [inline src/screens/Onboarding/OnboardingItem.tsx]: @AlphonceMutebi @Allen-Muhani @Chegemartinx 
The background color should not be specified here.
Should be in a central file.

Also the fontsize (line 47), fontweight (line 48), and so on, should not be defined here.

File: src/styles/theme.tsx [hunk 9]
```diff
@@ -3,11 +3,12 @@ const { width, height } = Dimensions.get("window");
 
 export const COLORS = {
   // base colors
-  primary: "#4840BB",
```
Comment (steffenkrogmann) [inline src/styles/theme.tsx:6]: @Allen-Muhani @AlphonceMutebi @Chegemartinx 
seems the "primary" base color, and bin line 22 "textBlack" both got replaced. Is that correct???

**Review by steffenkrogmann — CHANGES_REQUESTED** [review]
@Allen-Muhani @Alphonce-Mutebi @Chegemartinx 

I added a few in line comments. You might have to discuss them together.
(only based on my limited understanding).

File: src/screens/Drawer/CustomDrawer.tsx (hunk 1 above)
Comment (Allen-Muhani) [reply src/screens/Drawer/CustomDrawer.tsx]: Done

File: src/screens/Drawer/CustomDrawer.tsx (hunk 2 above)
Comment (Allen-Muhani) [reply src/screens/Drawer/CustomDrawer.tsx:49]: It is specific to that component. So I am of the opinion that it stays as is.

File: src/screens/Drawer/CustomDrawer.tsx (hunk 3 above)
Comment (Allen-Muhani) [reply src/screens/Drawer/CustomDrawer.tsx:153]: Bold is specific to the use case/context.

File: src/screens/Drawer/CustomDrawer.tsx (hunk 4 above)
Comment (Allen-Muhani) [reply src/screens/Drawer/CustomDrawer.tsx:156]: Bold is specific to the use case/context.

File: src/screens/Drawer/MyDrawer.tsx (hunk 7 above)
Comment (Allen-Muhani) [reply src/screens/Drawer/MyDrawer.tsx]: Done

File: src/screens/Drawer/MyDrawer.tsx (hunk 6 above)
Comment (Allen-Muhani) [reply src/screens/Drawer/MyDrawer.tsx]: Done

File: src/screens/Drawer/CustomDrawer.tsx (hunk 2 above)
Comment (Alphonce-Mutebi) [reply src/screens/Drawer/CustomDrawer.tsx:49]: @steffenkrogmann  seems ok ....to me

**Review by Alphonce-Mutebi — APPROVED** [review]

**Pull request merged**